packages/core/src/extensions/SideMenu/SideMenuPlugin.ts

```diff
@@ -411,29 +323,242 @@ export class SideMenuView<
   };
 
   /**
-   * If the event is outside the editor contents,
-   * we dispatch a fake event, so that we can still drop the content
-   * when dragging / dropping to the side of the editor
+   * Finds the closest editor visually to the given coordinates
+   */
+  private findClosestEditorElement = (coords: {
+    clientX: number;
+    clientY: number;
+  }) => {
+    // Get all editor elements in the document
+    const editors = Array.from(this.pmView.root.querySelectorAll(".bn-editor"));
+
+    if (editors.length === 0) {
+      return null;
+    }
+
+    // Find the editor with the smallest distance to the coordinates
+    let closestEditor = editors[0];
+    let minDistance = Number.MAX_VALUE;
+
+    editors.forEach((editor) => {
+      const rect = editor.getBoundingClientRect();
+
+      const distanceX =
+        coords.clientX < rect.left
+          ? rect.left - coords.clientX
+          : coords.clientX > rect.right
+            ? coords.clientX - rect.right
+            : 0;
+
+      const distanceY =
+        coords.clientY < rect.top
+          ? rect.top - coords.clientY
+          : coords.clientY > rect.bottom
+            ? coords.clientY - rect.bottom
+            : 0;
+
+      const distance = Math.sqrt(
+        Math.pow(distanceX, 2) + Math.pow(distanceY, 2),
+      );
+
+      if (distance < minDistance) {
+        minDistance = distance;
+        closestEditor = editor;
+      }
+    });
+
+    return {
+      element: closestEditor,
+      distance: minDistance,
+    };
+  };
+
+  /**
+   * This dragover event handler listens at the document level,
+   * and is trying to handle dragover events for all editors.
+   *
+   * It specifically is trying to handle the following cases:
+   *  - If the dragover event is within the bounds of any editor, then it does nothing
+   *  - If the dragover event is outside the bounds of any editor, but close enough (within 250px) to the closest editor,
+   *    then it dispatches a synthetic dragover event to the closest editor (which will trigger the drop-cursor to be shown on that editor)
+   *  - If the dragover event is outside the bounds of the current editor, then it will dispatch a synthetic dragleave event to the current editor
+   *    (which will trigger the drop-cursor to be removed from the current editor)
+   *
+   * The synthetic event is a necessary evil because we do not control prosemirror-dropcursor to be able to show the drop-cursor within the range we want
```

Comment:
would it be a lot easier if we fork dropcursor?

normally I'd not suggest this, but given we already have an [alternative](packages/xl-multi-column/src/extensions/DropCursor/MultiColumnDropCursorPlugin.ts) dropcursor plugin for mult-column, it might be a cleaner solution?

Reply:
It would be cleaner to implement it like that... I'm just not sure it is worth the effort right now.